packages/block-library/src/query/edit/inspector-controls/index.js

```diff
@@ -178,14 +187,13 @@ export default function QueryInspectorControls( props ) {
 		showParentControl ||
 		showFormatControl;
 	const dropdownMenuProps = useToolsPanelDropdownMenuProps();
-
+	const hasPatterns = !! usePatterns( clientId, name ).length;
```

Comment:
The proper check for existing patterns is by using `useBlockNameForPatterns`. If for some reason we don't have `core/query` patterns but we have patterns for a specific Query Loop variation they wouldn't show.